MapboxMobileEvents/MMEEventsManager.m

```diff
@@ -202,6 +205,10 @@ - (void)flush {
     
     NSArray *events = [self.eventQueue copy];
     NSUInteger eventsCount = events.count;
+    
```

Comment:
I'd argue that cleaning the queue should still be part of `flush()` otherwise it would only be posting the events. Also, for debug events -- these are already being called on a separate method. I think inline is the cleanest way to do this and provides some additional documentation for developers freshly entering this project. 

I agree this method has gotten a bit bloated. I'll see how we can make this more readable. Thanks for bringing this up 👍